src/Raven.Studio/typescript/common/helpers/graph/rangeAggregator.ts

```diff
@@ -3,73 +3,94 @@
 import d3 = require("d3");
 
 class rangeAggregator {
-    
-    aggregation = [] as Array<aggregatedRange>;
-
-    pushRange(start: number, end: number) {
-        if (this.aggregation.length === 0) {
-            this.aggregation.push({
-                start: start,
-                end: end,
-                value: 1
-            });
-            return;
-        }
 
-        // before all
-        if (end < this.aggregation[0].start) {
-            if (this.aggregation[0].end !== start) {
-                this.aggregation.unshift({
-                    start: this.aggregation[0].end,
-                    end: start,
-                    value: 0
-                });
-            }
+    items: Array<workTimeUnit> = [];
+    indexesWork: Array<indexesWorkData> = [];
+    ptrStart: number;
+    ptrEnd: number;
+    maxConcurrentIndexes: number;
 
-            this.aggregation.unshift({
-                start: start,
-                end: end,
-                value: 1
-            });
-            return;
-        }
+    constructor(inputItems: Array<[Date, Date]>) {
+        this.maxConcurrentIndexes = 0;
 
-        // after all
-        if (start > this.aggregation.last().end) {
-            if (this.aggregation.last().end !== start) {
-                this.aggregation.push({
-                    start: this.aggregation.last().end,
-                    end: start,
-                    value: 0
-                });
-            }
+        this.items = inputItems.map((x) => { return { startTime: x[0].getTime(), endTime: x[1].getTime() } });
```

Comment:
inputItems.map((x) => ({ startTime: x[0].getTime(), endTime: x[1].getTime() }));

you have to enclose object in () to make it work.